job-service-db/src/main/resources/procedures/getJobs.sql

```diff
@@ -183,6 +184,46 @@ BEGIN
          ELSE quote_ident(in_sort_field)
        END ||
         ' ' || CASE WHEN in_sort_ascending THEN 'ASC' ELSE 'DESC' END;
-    RETURN QUERY EXECUTE sql;
+
+    -- Create temporary table as a base to update the job progress
+    EXECUTE 'CREATE TEMPORARY TABLE get_job_temp ON COMMIT DROP AS ' || sql;
+
+    ALTER TABLE get_job_temp ADD COLUMN id SERIAL PRIMARY KEY;
+
+    -- Create a duplicate of that temporary table to store the results after the update
+    CREATE TEMPORARY TABLE new_table ON COMMIT DROP AS SELECT * FROM get_job_temp ORDER BY id;
+
+    FOR jobId IN SELECT * FROM get_job_temp LOOP
+        -- Take out an exclusive update lock on the job row
+        PERFORM NULL FROM job j
+        WHERE j.partition_id = in_partition_id
+            AND j.job_id = jobId
+        FOR UPDATE;
+
+        -- Process outstanding job updates
+        PERFORM internal_update_job_progress(in_partition_id, jobId);
+        UPDATE new_table nt SET
+               status = j.status,
+               percentage_complete = j.percentage_complete
+        FROM job j
+        WHERE nt.job_id = j.job_id;
+    END LOOP;
+
+    -- Return the new table created
+    RETURN QUERY
+    SELECT at.job_id,
```

Comment:
What is the point of the `at.` alias here?  And if there is a point of it, why doesn't `value` have it?

Reply:
it prevents the following error:
{
  "message": "ERROR: column reference \"job_id\" is ambiguous\n  Detail: It could refer to either a PL/pgSQL variable or a table column.\n  Where: PL/pgSQL function get_jobs(character varying,character varying,character varying,integer,integer,character varying,character varying,boolean,character varying[],character varying) line 140 at RETURN QUERY"
}
It seems to be conflicting with the job_id field from the table we return.
I think it could be used only on the

Reply:
Ok